test/revise-spec.ts

```diff
@@ -151,4 +151,64 @@ describe('revise', () => {
     assert.notStrictEqual(
         diff3Obj.baz, source.baz, 'Should return new path for changed value');
   });
+
+  describe('when circular references are present', () => {
+    it('handles self pointing circular references', () => {
+      const source: any = {};
+      source.foo = source;
+      const revision: any = {};
+      revision.foo = revision;
+
+      const newSource = revise(source, revision);
+      assert.strictEqual(newSource.foo, revision);
+    });
+
+    it('handles reversed pointing circular references', () => {
+      const source: any = {};
+      const revision: any = {};
+      source.foo = revision;
+      revision.foo = source;
+
+      const newSource = revise(source, revision);
+
+      // Strange case, it ends up being source..
+      assert.strictEqual(newSource.foo, revision);
```

Comment:
@wcjordan This assert ended up changing from 
assert.strictEqual(newSource.foo, source); 
to 
assert.strictEqual(newSource.foo, revision);

Its strange but I think the logic works out?

Reply:
Ya the idea is we retain the source object right?  so the original source.foo is revision.